src/components/List/List.jsx

```diff
@@ -1,35 +1,74 @@
-import React, { useRef, forwardRef, useMemo } from "react";
+import React, { useRef, forwardRef, useMemo, useState, useCallback } from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
 import useMergeRefs from "../../hooks/useMergeRefs";
 import { VirtualizedListItems } from "components/List/VirtualizedListItems/VirtualizedListItems";
 import "./List.scss";
+import { keyCodes } from "../../constants/KeyCodes";
 
 const List = forwardRef(
   ({ className, id, component, children, dense, ariaLabel, ariaDescribedBy, renderOnlyVisibleItems, style }, ref) => {
     const componentRef = useRef(null);
+    const [focusIndex, setFocusIndex] = useState(0);
     const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
     const Component = component;
+    const childrenRefs = useRef([]);
+    const changeFocus = focusIndex => {
+      setFocusIndex(focusIndex);
+      childrenRefs.current[focusIndex].focus();
+    };
+    const onKeyDown = useCallback(
+      event => {
+        const isUpKey = event.keyCode === keyCodes.UP_ARROW;
+        const isDownKey = event.keyCode === keyCodes.DOWN_ARROW;
+        if (isUpKey || isDownKey) {
+          if (isDownKey && focusIndex + 1 < childrenRefs.current.length) {
+            changeFocus(focusIndex + 1);
```

Comment:
you can add to this function a overrideFocusIndex = undef and inside every condition only change it's value.
after the condition you can check if overrideFocusIndex!== undefined and only if it is you can call the set focus logic (not in a function)